File: May-LeetCoding-Challenge/04-Number-Complement/Number-Complement.java
class Solution {
    public int findComplement(int num) {
        int numBits = 0, origin = num;
        
        // get number of bits in binary representation
        while (num > 0){
            num >>= 1;
            numBits++;
        }

        return ((1 << numBits) - 1) ^ origin;
    }
}

File: May-LeetCoding-Challenge/04-Number-Complement/Number-Complement.js
/**
 * @param {number} num
 * @return {number}
 */
// Solution 1: 
var findComplement = function(num) {
    var i = 0;
    var count = 0;
    while (num>0){
        if((num&1)==0){
            i += (1<<count)
        }
        count++;
        num = (num>>1);
    }
    return i;
};
